### PR TITLE
added ReasonableFeatureExtractionSettings

### PR DESCRIPTION
**Pull request by MaxBenChrist**
please do not merge this yet. there are still some unit tests missing for the ReasonableFeatureExtractionSettings object.

this will fix the failing unit tests, the build error and issue #122

### CODE REVIEW

**Comment by MaxBenChrist** [comment]
added unit tests. this will repair the broken build

File: tsfresh/feature_extraction/settings.py [hunk 1]
```diff
@@ -357,4 +352,27 @@ def __init__(self):
...
                 name_to_param_copy[feature_calculator] = self.name_to_param[feature_calculator]
 
-        self.name_to_param = name_to_param_copy
\ No newline at end of file
+        self.name_to_param = name_to_param_copy
+
+
+class ReasonableFeatureExtractionSettings(FeatureExtractionSettings):
+    """
+    This class is a parent class of the FeatureExtractionSettings class
```
Comment (jneuff) [inline tsfresh/feature_extraction/settings.py]: typo: child class

**Comment by nils-braun** [comment]
Should we actually make this the default?

**Comment by earthgecko** [comment]
@nils-braun :+1: 

I spent a few hours debugging why timeseries with very little variability take so long.
The example timeseries below (all 15.0, except the last datapoint) is taking ~578 seconds
to extract features from, where as mostly all other timeseries of the same length
are taking > 7 seconds.

```
timestamps = []
values = []
count = 0
ts = 1482644940
while count < 10075:
    count += 1
    timestamps.append(ts)
    values.append(15.0)
    ts += 60

timeseries = map(list, zip('t' * 10075, timestamps, values))
timeseries[-1][2] = 16.0

df = pd.DataFrame(timeseries, columns=['metric', 'timestamp', 'value'])
df_features = extract_features(df, column_id='metric', column_sort='timestamp', column_kind=None, column_value=None)
```

Using the new ReasonableFeatureExtractionSettings solved these edge cases, ~1.78 seconds with the loss of 6 features, so 210 features instead of 216.

```
tsf_settings = ReasonableFeatureExtractionSettings()
# Disable tqdm progress bar
tsf_settings.disable_progressbar = True
df_features = extract_features(
    df, column_id='metric', column_sort='timestamp', column_kind=None,
    column_value=None, feature_extraction_settings=tsf_settings)
```

I am very happy that @MaxBenChrist added ReasonableFeatureExtractionSettings and
I would say that it would probably be useful as default as it is going to catch
a lot of edge cases, like this type of timeseries described above.

It suggests that one or more of the tsfresh algorithms are calculating using
a sequence which is being appended for every same value in sequence and then
calculating the result and repeating the same operation over and over.  In most
timeseries this/these algorithms would probably break out of the sequence and
reset as soon as the next value was different from the last?

**Comment by MaxBenChrist** [comment]
> I am very happy that @MaxBenChrist added ReasonableFeatureExtractionSettings and
I would say that it would probably be useful as default as it is going to catch
a lot of edge cases, like this type of timeseries described above.

As we have a semantic versioning scheme, this could be interpreted as a major change (a major version is making incompatible API changes). This is something I would prefer to not do at the moment.

> It suggests that one or more of the tsfresh algorithms are calculating using
a sequence which is being appended for every same value in sequence and then
calculating the result and repeating the same operation over and over. In most
timeseries this/these algorithms would probably break out of the sequence and
reset as soon as the next value was different from the last?

I am not sure what you are proposing here, @earthgecko. By the way, you can use constructions like [15.0] *1000 to get your time series lists faster.

**Comment by earthgecko** [comment]
@MaxBenChrist quite sensible considering semantic versioning, perhaps maybe be
suitable for v1.x if it is considered overall beneficial.

Well on the culprit algorithm/s it is not a bug, I am sure that calculations are
correct, I was just wondering if it would be possible to optimise the running of
sample_entropy or approximate_entropy any further.  Not optimizing the algorithms
in any way, just adding some sanity checks to assess if they should run or not.
That may be non-trivial in some case and result in "skewed" im terms of repeatability,
if an algorithm is only run if not some_condition, which would create a state.
For example if a timeseries has a certain "shape" shapes defined by certain
features e.g. range, unique values, etc then do not run x high_comp_cost algorithm.

Not desirable.  Not semver but perhaps something that may be useful in the future.

However that said, I would say that the cost of running to high_comp_cost
algorithms on the edge cases is exorbitantly high, given the possible timeseries
that are being thrown at tsfresh.  There have been people that have run into
long runtimes and people are going to encounter these edge cases.

So if the API cannot change, the documentation could now promote the use of 
ReasonableFeatureExtractionSettings a bit more and specifically describe some
now known possible pitfalls that may be encountered running the default
settings on extract_features, unless long runtimes and a times very variable in
runtimes can be expected and handled.  Not all timeseries created equally :)

On the "you can use constructions" I normally do, but I did not want to have to
upload a csv timeseries file with the example and match the data and method to
reproduce injecting the metric string into the map constructor seemed problematic

```
timeseries = map(list, zip('t' * 10075, map(int, range(int(ts) - 604740, int(ts) + 60)), [15.0] * 10075))
```

Does not behave the same I tried and hammered around a bit with it and then I
just decide to get it **done** instead of trail and error how to map it :)

**Pull request merged**